source/structured-streaming.txt

```diff
@@ -0,0 +1,327 @@
+.. _spark-structured-streaming:
+
+=================================
+Structured Streaming with MongoDB
+=================================
+
+.. default-domain:: mongodb
+
+.. contents:: On this page
+   :local:
+   :backlinks: none
+   :depth: 2
+   :class: singlecol
+
+Overview
+--------
+
+Spark Structured Streaming is a data stream processing engine you can 
+use through the Dataset or DataFrame API. The MongoDB Spark Connector 
+enables you to stream to and from MongoDB using Spark Structured 
+Streaming.
+
+.. include:: includes/streaming-distinction.rst
+
+To learn more about Structured Streaming, see the 
+`Spark Programming Guide
+<https://spark.apache.org/docs/latest/structured-streaming-programming-guide.html>`__.
+
+.. _write-structured-stream:
+
+Configuring a Write Stream to MongoDB
+-------------------------------------
+
+Specify write stream configuration settings on your streaming 
+Dataset or DataFrame with ``.writeStream``.
+
+You must specify the following configuration settings to write to 
+MongoDB:
+
+.. list-table::
+   :header-rows: 1
+   :stub-columns: 1
+   :widths: 10 40
+
+   * - Setting
+     - Description
+
+   * - ``.writeStream.format()``
+     - The format to use for write stream data. Use "mongodb".
+
+   * - ``.writeStream.option()``
+     - Use the ``.option`` method to specify your MongoDB deployment 
+       connection string with the "spark.mongodb.connection.uri" 
+       option key.
+
+       You must specify a database and collection, either as part of 
+       your connection string or with additional ``.option`` 
+       methods using the following keys:
+
+       - "spark.mongodb.database"
+       - "spark.mongodb.collection"
+
+   * - ``.writeStream.outputMode()``
+     - The output mode to use.
```

Comment:
Suggestion: Could a link be provided to somewhere where I could see what output modes I can choose from?

```suggestion
     - The output mode to use. To view a list of all supported output modes, see `<some link>`__.
```

Reply:
This comment made me realize I need to move a lot of this into a dedicated python tab - anywhere I link to pyspark docs or reference pyspark methods should be under a tab.

Updated.